api/src/main/java/jakarta/mail/BodyPart.java

```diff
@@ -39,13 +39,6 @@ public abstract class BodyPart implements Part {
      */
     protected Multipart parent;
 
-    /**
-     * Instance of stream provider.
-     *
-     * @since JavaMail 2.1
-     */
-    protected final StreamProvider streamProvider = StreamProvider.provider();
```

Comment:
The more I think about it, `Session.getDefaultInstance(System.getProperties(), null).getStreamProvider();` is going to pin the classloader of the implementation which is not going to play nice when implementation classloader is shutdown.  Which in a sense is a whole new issue.

That said, should we keep this field, remove final, maybe add volatile, and lazy set it to `StreamProvider.provider();` when we need to?  We shouldn't even need any locking as the race is benign.

Reply:
I would like to be consistent with this. If we decide to not depend on `Session.getDefaultInstance(System.getProperties(), null).getStreamProvider();` we should carry this decision to the other parts. This means to apply your suggestion in more places where StreamProvider could be null.

Do I implement it?.

Reply:
Ok. Let's keep default session logic to keep this ticket simple.  I'll do some test to see if default session breaks if implementation loader is shutdown. We can deal with that in a new ticket if needed.